bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java

```diff
@@ -497,6 +492,11 @@ private void updateLightState(NhcAction2 action, List<NhcProperty> devicePropert
                 logger.debug("setting action {} internally to {}", action.getId(), dimmerProperty.get().brightness);
             }
         }
+
+        if ((booleanState != null) && NHCON.equals(booleanState)) {
```

Comment:
You can suppress your first test condition as it is already covered by your second condition.